shepherd_bms/src/stateMachine.cpp

```diff
@@ -213,177 +193,84 @@ void StateMachine::requestTransition(BMSState_t next_state)
 }
 
 
-uint32_t StateMachine::faultCheck(AccumulatorData_t *accData)
+uint32_t StateMachine::faultReturn(AccumulatorData_t *accData)
 {
 	// FAULT CHECK (Check for fuckies)
-	uint32_t faultStatus = 0;
 
-	//prefault for Low Cell Voltage
-
-	if (prefaultLowCell.faultEvalState == BEFORE_TIMER_START && accData->min_voltage.val < MIN_VOLT * 10000)
-    {
-        prefaultLowCell.faultTimer.startTimer(PRE_UNDER_VOLT_TIME);
-        prefaultLowCell.faultEvalState = DURING_FAULT_EVAL;
-    }
-    else if (prefaultLowCell.faultEvalState == DURING_FAULT_EVAL)
-    {
-        if (prefaultLowCell.faultTimer.isTimerExpired())
-        {
-            if (prefaultCANDelay1.isTimerExpired())
-			{
-            	compute.sendDclPreFault(true);
-				prefaultCANDelay1.startTimer(CAN_MESSAGE_WAIT);
-			}
-        }
-        if (!(accData->min_voltage.val < MIN_VOLT * 10000))
+	struct fault_eval fault_table[8] = 
```

Comment:
1. i made it static at first but then google told me structs cant/shouldnt be static but i can change back, I couldnt make it a private member in state machine easily because state machine as a class doesnt have access to current BMSdata which is needed to fill in param

2. same as my note above, i agree that its too long but it lowkey seemed messier when i spread it across multiple, but will spend more time messing with it to try and find best

Reply:
lmao im a dumbass thats only for defining a struct ignore 1